### PR TITLE
Integration test for the social example.

### PR DESCRIPTION
**Pull request by MathieuDutSik**
This is the integration test for the social media example app.

### CODE REVIEW

File: examples/social/tests/cross_chain.rs [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+    // Now make chain1 handle that fact.
+
+    chain1.handle_received_effects().await;
+
+    // Querying the own posts
+
+    let query_string = "query { ownPosts(start: 0, end:1) { timestamp, text } }";
+    let value: serde_json::Value = chain2.query(application_id, query_string).await;
+    let value = value["data"]["ownPosts"][0]["text"].clone();
+    assert_eq!(value, "Linera is the new mastodon".to_string());
```
Comment (afck) [inline examples/social/tests/cross_chain.rs:60]:
```suggestion
    assert_eq!(value, "Linera is the new Mastodon".to_string());
```

File: examples/social/tests/cross_chain.rs [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+            );
+        })
+        .await;
+
+    // Now make chain1 handle that fact.
+
+    chain1.handle_received_effects().await;
+
+    // Querying the own posts
+
```
Comment (ma2bd) [inline examples/social/tests/cross_chain.rs:56]: nit: We don't need as many newlines everywhere, especially when the comment applies to what immediately follows.

File: examples/social/tests/cross_chain.rs [hunk 3]
```diff
@@ -0,0 +1,68 @@
+// Copyright (c) Zefchain Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+//! Integration tests for the Fungible Token application.
```
Comment (ma2bd) [inline examples/social/tests/cross_chain.rs:4]: social?

File: examples/social/tests/cross_chain.rs [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+#![cfg(not(target_arch = "wasm32"))]
+
+//use async_graphql::InputType;
+use linera_sdk::test::TestValidator;
+use social::Operation;
+
+/// Test posting messages across microchains.
+///
+/// Creates the application on chain2 indirectrly from the subscription, then
+/// send a message to chain2 and see it receiced on chain1.
```
Comment (ma2bd) [inline examples/social/tests/cross_chain.rs:15]: typo: receiced

Comment (ma2bd) [reply examples/social/tests/cross_chain.rs:15]: also: sends

File: examples/social/tests/cross_chain.rs (hunk 2 above)
Comment (MathieuDutSik) [reply examples/social/tests/cross_chain.rs:56]: Ok.

File: examples/social/tests/cross_chain.rs (hunk 3 above)
Comment (MathieuDutSik) [reply examples/social/tests/cross_chain.rs:4]: Yes.

File: examples/social/tests/cross_chain.rs [hunk 5]
```diff
@@ -0,0 +1,68 @@
+// Copyright (c) Zefchain Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+//! Integration tests for the Fungible Token application.
+
+#![cfg(not(target_arch = "wasm32"))]
+
+//use async_graphql::InputType;
```
[the remark this answers is not in the document]
Comment (MathieuDutSik) [reply examples/social/tests/cross_chain.rs:8]: Yes.

File: examples/social/tests/cross_chain.rs [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+use linera_sdk::test::TestValidator;
+use social::Operation;
+
+/// Test posting messages across microchains.
+///
+/// Creates the application on chain2 indirectrly from the subscription, then
```
[the remark this answers is not in the document]
Comment (MathieuDutSik) [reply examples/social/tests/cross_chain.rs:14]: Yes.

File: examples/social/tests/cross_chain.rs (hunk 4 above)
Comment (MathieuDutSik) [reply examples/social/tests/cross_chain.rs:15]: Done.

**Review by ma2bd — APPROVED** [review]

**Pull request merged**